alibi_detect/utils/keops/kernels.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from pykeops.torch import LazyTensor
+import torch
+import torch.nn as nn
+from typing import Callable, Optional
+
+
+def sigma_mean(x: LazyTensor, y: LazyTensor, dist: LazyTensor) -> torch.Tensor:
```

Comment:
Curious why `torch` and `tensorflow` backends have `sigma_median` heuristic, but this one has `sigma_mean`?

Reply:
As far as I know it is not possible to easily infer the median pairwise distance using KeOps (not a reduction operation on the symbolic tensor). As a result I opted for the mean instead of median pairwise distance. Realistically either could work well/terrible for a given use case.